agent/stats/engine.go

```diff
@@ -414,7 +414,7 @@ func (engine *DockerStatsEngine) addToStatsContainerMapUnsafe(
 // GetInstanceMetrics gets all task metrics and instance metadata from stats engine.
 func (engine *DockerStatsEngine) GetInstanceMetrics(includeServiceConnectStats bool) (*ecstcs.MetricsMetadata, []*ecstcs.TaskMetric, error) {
 	var taskMetrics []*ecstcs.TaskMetric
-	idle := engine.isIdle()
+	idle := engine.isIdle() && engine.serviceConnectIsIdle()
```

Comment:
should this be `||` instead of `&&`

Reply:
`idle` is true only when both container insights are disabled and there are not service connect metrics to send, otherwise TACS need the request to be non-idle.